patches/tModLoader/Terraria/ModLoader/Config/ConfigManager.cs

```diff
@@ -72,6 +74,113 @@ internal static void Add(ModConfig config)
 		configList2.Add(GeneratePopulatedClone(config));
 	}
 
+	internal static void FinishSetup()
+	{
+		// Register localization for all fields and properties that should show
+		foreach (var activeConfigs in ConfigManager.Configs) {
+			foreach (var config in activeConfigs.Value) {
+				try {
+					string modConfigLabelKey = GetModConfigLabelKey(config, throwErrors: true);
+					Language.GetOrRegister(modConfigLabelKey, () => Regex.Replace(config.Name, "([A-Z])", " $1").Trim());
+				}
+				catch (ValueNotTranslationKeyException e) {
+					e.additional = $"The ModConfig class '{config.Name}' caused this exception.";
+					e.Data["mod"] = config.Mod.Name;
+					throw;
+				}
+
+				RegisterLocalizationKeysForMembers(config, config.GetType());
+			}
+		}
+	}
+
+	private static void RegisterLocalizationKeysForMembers(ModConfig config, Type type)
+	{
+		foreach (PropertyFieldWrapper variable in ConfigManager.GetFieldsAndProperties(type)) {
+#pragma warning disable CS0618
+			if (Attribute.IsDefined(variable.MemberInfo, typeof(JsonIgnoreAttribute)) && !(Attribute.IsDefined(variable.MemberInfo, typeof(LabelAttribute)) || Attribute.IsDefined(variable.MemberInfo, typeof(ShowDespiteJsonIgnoreAttribute))))
+				continue;
+#pragma warning restore CS0618
+
+			try {
+				// TODO: Classes used in Generic types and Enums
+				// TODO: throw error after updating hjson somehow, so auto keys are populated?
```

Comment:
What does this mean?

Reply:
Hm, this meant the modder would see a half fixed hjson, even if they had a malformed attribute. It's not needed, modders should just follow the porting instructions I posted and do it in 1 pass. I'll remove the comment